### ISSUE TITLE
Configuration option for disabling self-registration

### ISSUE THREAD

**Issue by mgrothe-gu** [DSpace/dspace-angular#1153]
**Is your feature request related to a problem? Please describe.**
Self registration of users can not be disabled in DSpace 7 Beta 5.0. Most institutions mange user accounts via services like LDAP, but also require password-authentication to be enabled for local admin accounts. With local accounts enabled, registration with an email address is enabled too. The prominent sign-up field might convince members of the institution visiting DSpace for the first time, that they need to register.

**Describe the solution you'd like**
It should be possible to disable self-registration.

**Describe alternatives or workarounds you've considered**
Setting the valid domains for self-signup to a random value prevents successful registration, but doesn't get rid of the confusion of users and resulting support cases when their attempt to register fails.

**Comment by tdonohue** [comment]
Agreed, this should be implemented in 7.x...it'll likely need to wait for 7.1 though (tentatively scheduling for that release). 

For implementers, this feature previously existed in v6.x in the XMLUI as the `xmlui.user.registration` configuration (in `dspace.cfg`).  It likely could be implemented similarly with a new backend configuration (I suspect it would need to be on the backend to ensure registration is disabled both from the UI and the REST API.)
